compiler/ast/reports.nim

```diff
@@ -224,6 +224,13 @@ func severity*(
       of repDebug:    report.debugReport.severity()
       of repExternal: report.externalReport.severity()
 
+func isFatalSeverity*(report: Report): bool =
```

Comment:
This procedure is not a general query as to whether something is fatal, it's specifically the compiler's viewpoint to treat the only internal and semantic errors as fatal, regardless of present and future code.

The general query is handled via the `severity` routine defined above.

Reply:
I consider this function to be of a different nature, unlike `isCompilerFatal`, it's not meant for deciding what the compiler does, as it merely reports whether the severity is fatal, nothing else. I added it because it may be slightly  more efficient than using `severity == rsevFatal`. I added this to the notes of the PR.

Reply:
Please remove `isFatalSeverity` and replace its usage with `severity == rsevFatal` the (very small) efficiency improvement is outweighed by the problematic introduction of another source of truth. If a new fatal non-sem/non-internal report is ever added, `isFatalSeverity` will silently stop doing what it should.

Reply:
Thanks.